unchanged:
--- b/build/pdf.js
+++ b/build/pdf.js
@@ -362,6 +362,11 @@
   INCORRECT_PASSWORD: 2
 };
 exports.PasswordResponses = PasswordResponses;
+const ViewerFontColor = '#000000';
+exports.ViewerFontColor = ViewerFontColor;
+const ViewerBgColor = '#ffffff';
+exports.ViewerBgColor = ViewerBgColor;
+
 let verbosity = VerbosityLevel.WARNINGS;
 function setVerbosityLevel(level) {
   if (Number.isInteger(level)) {
@@ -6797,7 +6802,7 @@
     const width = this.ctx.canvas.width;
     const height = this.ctx.canvas.height;
     const savedFillStyle = this.ctx.fillStyle;
-    this.ctx.fillStyle = background || "#ffffff";
+    this.ctx.fillStyle = background || _util.ViewerBgColor;
     this.ctx.fillRect(0, 0, width, height);
     this.ctx.fillStyle = savedFillStyle;
     if (transparency) {
@@ -7605,7 +7610,12 @@
       }
       if (this.contentVisible && (glyph.isInFont || font.missingFile)) {
         if (simpleFillText && !accent) {
+          const _style = ctx.fillStyle;
+          if (ctx.fillStyle==='#000000') {
+            ctx.fillStyle = _util.ViewerFontColor;
+          }
           ctx.fillText(character, scaledX, scaledY);
+          ctx.fillStyle = _style;
         } else {
           this.paintChar(character, scaledX, scaledY, patternTransform);
           if (accent) {
@@ -8888,6 +8898,7 @@
 GlobalWorkerOptions.workerSrc = "";
 
 /***/ }),
+
 /* 15 */
 /***/ ((__unused_webpack_module, exports, __w_pdfjs_require__) => {
 
@@ -17651,6 +17662,24 @@
     return _worker_options.GlobalWorkerOptions;
   }
 }));
+Object.defineProperty(exports, "ViewerFontColor", ({
+  enumerable: true,
+  get: function () {
+    return _util.ViewerFontColor;
+  },
+  set: function (color) {
+    _util.ViewerFontColor = color;
+  }
+}));
+Object.defineProperty(exports, "ViewerBgColor", ({
+  enumerable: true,
+  get: function () {
+    return _util.ViewerBgColor;
+  },
+  set: function (color) {
+    _util.ViewerBgColor = color;
+  }
+}));
 Object.defineProperty(exports, "ImageKind", ({
   enumerable: true,
   get: function () {
unchanged:
--- b/build/pdf.js
+++ b/build/pdf.js
@@ -6786,6 +6786,7 @@
     this._cachedScaleForStroking = [-1, 0];
     this._cachedGetSinglePixelWidth = null;
     this._cachedBitmapsMap = new Map();
+    this.opStack = [];
   }
   getObject(data, fallback = null) {
     if (typeof data === "string") {
@@ -6844,6 +6845,24 @@
         return i;
       }
       fnId = fnArray[i];
+
+      // const opName = Object.entries(_util.OPS).find(([key, value]) => value === fnId)[0];
+      switch(fnId) {
+        case 74: //'paintFormXObjectBegin'
+          this.opStack.push(fnId);
+          break;
+        case 75: //'paintFormXObjectEnd'
+          this.opStack.pop();
+          break;
+      }
+      if (fnId===58) { //'setStrokeRGBColor'
+        const enableCondition = this.opStack.length===0 && fnArray.length>10;
+        if (enableCondition && argsArray[i][0]===0 && argsArray[i][1]===0 && argsArray[i][2]===0) {
+          const parsedRGBColor = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(_util.ViewerFontColor).slice(1).map(n => parseInt(n, 16));
+          argsArray[i] = parsedRGBColor;
+        }
+      }
+
       if (fnId !== _util.OPS.dependency) {
         this[fnId].apply(this, argsArray[i]);
       } else {
only in patch2:
unchanged:
--- a/build/pdf.js
+++ b/build/pdf.js
@@ -6787,6 +6787,7 @@ class CanvasGraphics {
     this._cachedGetSinglePixelWidth = null;
     this._cachedBitmapsMap = new Map();
     this.opStack = [];
+    this.opTrace = [];
   }
   getObject(data, fallback = null) {
     if (typeof data === "string") {
@@ -6847,6 +6848,11 @@ class CanvasGraphics {
       fnId = fnArray[i];

       // const opName = Object.entries(_util.OPS).find(([key, value]) => value === fnId)[0];
+      // if (fnId===44) {
+      //   this.opTrace.push([ opName, argsArray[i][0].map(x => x.unicode).join('') ]);
+      // } else {
+      //   this.opTrace.push(opName);
+      // }
       switch(fnId) {
         case 74: //'paintFormXObjectBegin'
           this.opStack.push(fnId);
@@ -6855,8 +6861,12 @@ class CanvasGraphics {
           this.opStack.pop();
           break;
       }
+      const isInPainting = !(this.opStack.length===0);
+      if (fnId===44) { //'showText'
+        argsArray[i] = [ ...argsArray[i], isInPainting ];
+      }
       if (fnId===58) { //'setStrokeRGBColor'
-        const enableCondition = this.opStack.length===0 && fnArray.length>10;
+        const enableCondition = !isInPainting && fnArray.length>10;
         if (enableCondition && argsArray[i][0]===0 && argsArray[i][1]===0 && argsArray[i][2]===0) {
           const parsedRGBColor = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(_util.ViewerFontColor).slice(1).map(n => parseInt(n, 16));
           argsArray[i] = parsedRGBColor;
@@ -7524,7 +7534,7 @@ class CanvasGraphics {
     }
     return (0, _util.shadow)(this, "isFontSubpixelAAEnabled", enabled);
   }
-  showText(glyphs) {
+  showText(glyphs, isInPainting=false) {
     const current = this.current;
     const font = current.font;
     if (font.isType3Font) {
@@ -7630,7 +7640,7 @@ class CanvasGraphics {
       if (this.contentVisible && (glyph.isInFont || font.missingFile)) {
         if (simpleFillText && !accent) {
           const _style = ctx.fillStyle;
-          if (ctx.fillStyle==='#000000') {
+          if (ctx.fillStyle==='#000000' && !isInPainting) {
             ctx.fillStyle = _util.ViewerFontColor;
           }
           ctx.fillText(character, scaledX, scaledY);
@@ -11456,7 +11466,7 @@ class SVGGraphics {
           this.setFont(args);
           break;
         case _util.OPS.showText:
-          this.showText(args[0]);
+          this.showText(args[0], args[1]);
           break;
         case _util.OPS.showSpacedText:
           this.showText(args[0]);
